pkg/operator/controller/garden/reconciler.go

```diff
@@ -59,24 +63,60 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
+	conditionReconciled := v1beta1helper.GetOrInitConditionWithClock(r.Clock, garden.Status.Conditions, operatorv1alpha1.GardenReconciled)
+	conditionReconciled = v1beta1helper.UpdatedConditionWithClock(r.Clock, conditionReconciled, gardencorev1beta1.ConditionProgressing, conditionReasonPrefix(garden)+"Progressing", "Garden operation is currently being processed.")
```

Comment:
I think `status.observedGeneration` should be updated right here.
With this, `observedGeneration` would be up-to-date even when running into a reconciliation error. I believe this is semantically correct.